pkg/controller/attach_detach.go

```diff
@@ -81,7 +83,7 @@ var allowedTargetAttachmentStates = map[string][]string{
 	string(azdiskv1beta2.Detaching):          {string(azdiskv1beta2.Detached), string(azdiskv1beta2.DetachmentFailed)},
 	string(azdiskv1beta2.Attached):           {string(azdiskv1beta2.Detaching)},
 	string(azdiskv1beta2.Detached):           {},
-	string(azdiskv1beta2.AttachmentFailed):   {string(azdiskv1beta2.Detaching)},
+	string(azdiskv1beta2.AttachmentFailed):   {string(azdiskv1beta2.Detaching), string(azdiskv1beta2.AttachmentPending)},
```

Comment:
State transition from `AttachmentFailed` to `AttachmentPending` shouldn't be allowed as this would allow `attach_detach_controller` to retry on failed attachments on its own without external triggers. When we wish to update state from `AttachmentFailed` to `AttachmentPending` to retrigger attach operation in `replica controller`, we should use force update mode instead.

Reply:
Got it, thanks for the explanation.